core/src/main/java/apoc/nodes/Nodes.java

```diff
@@ -67,6 +77,51 @@
 
     @Context
     public Pools pools;
+    
+    @Procedure
+    @Description("CALL apoc.nodes.cycles([nodes], 'type', $config) - Detect all path cycles from node list")
+    public Stream<PathResult> cycles(@Name("nodes") List<Node> nodes, @Name(value = "config",defaultValue = "{}") Map<String, Object> config) {
+        NodesConfig conf = new NodesConfig(config);
+        final String type = conf.getRelType();
+        Stream<Path> paths = nodes.stream().flatMap(start -> {
+            boolean allRels = StringUtils.isBlank(type);
+            final RelationshipType relType = RelationshipType.withName(type);
+            final Iterable<Relationship> relationships = allRels
+                    ? start.getRelationships(Direction.OUTGOING)
+                    : start.getRelationships(Direction.OUTGOING, relType);
+            final PathExpander<Path> pathExpander = (allRels 
+                    ? PathExpanderBuilder.allTypes(Direction.OUTGOING) 
+                    : PathExpanderBuilder.empty().add(relType, Direction.OUTGOING)
+            ).build();
+            PathFinder<Path> finder = GraphAlgoFactory.shortestPath(
+                    new BasicEvaluationContext(tx, db),
+                    pathExpander,
+                    conf.getMaxDepth());
+            Map<Long, List<Long>> dups = new HashMap<>();
+            return Iterables.stream(relationships)
+                    // to prevent duplicated (start and end nodes with double-rels)
+                    .filter(relationship -> {
+                        final List<Long> nodeDups = dups.computeIfAbsent(relationship.getStartNodeId(), (key) -> new ArrayList<>());
+                        if (nodeDups.contains(relationship.getEndNodeId())) {
+                            return false;
+                        }
+                        nodeDups.add(relationship.getEndNodeId());
+                        return true;
+                    })
+                    .map(relationship -> {
```

Comment:
Could we use a flatmap here? Then return an empty stream if it's null, otherwise stream consisting of the path.